src/core/lib/iomgr/ev_poll_posix.cc

```diff
@@ -267,19 +267,29 @@ struct grpc_pollset_set {
 //
 
 static void fork_fd_list_remove_node(grpc_fork_fd_list* node) {
+  gpr_mu_lock(&fork_fd_list_mu);
```

Comment:
this seems like a change in logic

Reply:
Not really. `fork_fd_list_remove_grpc_fd` is a new replacement for the old  `fork_fd_list_remove_node` and all call sites for `fork_fd_list_remove_node` are updated to use `fork_fd_list_remove_grpc_fd`

Reply:
@drfloob / @eugeneo should review this

Reply:
I believe
```
if (track_fds_for_fork) {
```
might be needed so there's no overhead if fork support is not necessary

Reply:
^ `fork_fd_list_remove_grpc_fd` does. No one calls `fork_fd_list_remove_node` other than `fork_fd_list_remove_grpc_fd`